python/pyspark/pandas/frame.py

```diff
@@ -8828,22 +8846,154 @@ def describe(self, percentiles: Optional[List[float]] = None) -> "DataFrame":
         else:
             percentiles = [0.25, 0.5, 0.75]
 
-        formatted_perc = ["{:.0%}".format(p) for p in sorted(percentiles)]
-        stats = ["count", "mean", "stddev", "min", *formatted_perc, "max"]
+        if len(exprs_numeric) == 0:
```

Comment:
Can we change this condition to: `len(exprs_numeric) == 0 and len(exprs_non_numeric) > 0`, and `raise ValueError("Cannot describe a DataFrame without columns"` at the end of `else` in this if-else branch?